File: src/main/java/dev/slimevr/autobone/AutoBoneProcessType.java
package dev.slimevr.autobone;

import java.util.HashMap;
import java.util.Map;


public enum AutoBoneProcessType {
	NONE(0),
	RECORD(1),
	SAVE(2),
	PROCESS(3),
	APPLY(4);

	public final int id;

	private static final Map<Integer, AutoBoneProcessType> byId = new HashMap<>();

	private AutoBoneProcessType(int id) {
		this.id = id;
	}

	public static AutoBoneProcessType getById(int id) {
		return byId.get(id);
	}

	static {
		for (AutoBoneProcessType abpt : values()) {
			byId.put(abpt.id, abpt);
		}
	}
}

File: server/src/main/java/dev/slimevr/autobone/AutoBoneProcessType.kt
package dev.slimevr.autobone

enum class AutoBoneProcessType(val id: Int) {
	NONE(0), RECORD(1), SAVE(2), PROCESS(3), APPLY(4);

	companion object {
		fun getById(id: Int): AutoBoneProcessType? = byId[id]
	}
}

private val byId = AutoBoneProcessType.values().associateBy { it.id }
